File: day5_part1_2015.py
with open("input.txt", "r") as file:
    strings = file.read().splitlines()

nice_count = 0

for s in strings:
    vowels = sum([1 for c in s if c in "aeiou"])
    double_letter = any([s[i] == s[i+1] for i in range(len(s)-1)])
    forbidden_strings = ["ab", "cd", "pq", "xy"]
    forbidden = any([fs in s for fs in forbidden_strings])
    
    if vowels >= 3 and double_letter and not forbidden:
        nice_count += 1

print(nice_count)

File: objectivec/day5_part1_2015.m
#import <Foundation/Foundation.h>

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        NSError *error;
        NSString *input = [NSString stringWithContentsOfFile:@"input.txt" encoding:NSUTF8StringEncoding error:&error];
        NSArray *strings = [input componentsSeparatedByString:@"\n"];
        
        int niceCount = 0;
        
        for (NSString *string in strings) {
            if ([string rangeOfString:@"ab"].location != NSNotFound || [string rangeOfString:@"cd"].location != NSNotFound || [string rangeOfString:@"pq"].location != NSNotFound || [string rangeOfString:@"xy"].location != NSNotFound) {
                continue;
            }
            
            NSCharacterSet *vowels = [NSCharacterSet characterSetWithCharactersInString:@"aeiou"];
            NSCharacterSet *doubleLetters = [NSCharacterSet characterSetWithCharactersInString:@"abcdefghijklmnopqrstuvwxyz"];
            BOOL hasThreeVowels = [[string componentsSeparatedByCharactersInSet:vowels] count] >= 4;
            BOOL hasDoubleLetter = NO;
            
            for (int i = 0; i < string.length - 1; i++) {
                if ([string characterAtIndex:i] == [string characterAtIndex:i + 1]) {
                    hasDoubleLetter = YES;
                    break;
                }
            }
            
            if (hasThreeVowels && hasDoubleLetter) {
                niceCount++;
            }
        }
        
        printf("%d\n", niceCount);
    }
    return 0;
}